rcl/src/rcl/context.c

```diff
@@ -126,6 +126,19 @@ __cleanup_context(rcl_context_t * context)
       }
     }
 
+    // clean up rmw_context
+    if (NULL != context->impl->rmw_context.impl) {
```

Comment:
IIUC this check means `rmw_context_fini()` won't be called for any current rmw implementation because they intentionally set `impl` to `NULL`. Maybe this could be a problem if an rmw_implementation uses global state instead of the `impl` pointer, but also expected cleanup code in `rmw_context_fini` to run?

Reply:
This is a valid point, I'll have to rethink how I avoid calling fini when the case is that init failed rather than a normal init-fini cycle.

Reply:
Makes sense to me